### PR TITLE
Issue #169 cannot set time for the future game in playoff

### LINKED ISSUE #169: Cannot set time for the future game in playoff tournament
When I want to set game time to the game where teams are not defined yet - I got following error:

image

### PR DESCRIPTION
**Pull request by MikolaB**
### Summary

Modify teams validation in game for Play Off scheme.

closes #169

#### Design

- [ ] Web API layer contains as little logic as possible
- [ ] Domain objects use semantic names

#### Perfromance

- [ ] In case of any DB query was changed: attach generated SQL for review
- [ ] Number of DB roundtrips should be as low as possible

#### Unit Tests

- [ ] Naming: Initial state and expected result are properly stated
- [ ] Test follows [AAA](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) pattern.
- [ ] Test is Isolated
- [ ] Mock instances and expected instances do not share references

### CODE REVIEW

File: src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs [hunk 1]
```diff
@@ -339,9 +339,17 @@ private void ValidateResult(Result result)
             ValidateSetScoresOrder(result.SetScores);
         }
 
-        private void ValidateTeams(int? homeTeamId, int? awayTeamId)
+        private void ValidateTeams(int? homeTeamId, int? awayTeamId, TournamentScheduleDto tournamentScheduleInfo)
         {
-            if (GameValidation.AreTheSameTeams(homeTeamId, awayTeamId))
+            if (tournamentScheduleInfo.Scheme != TournamentSchemeEnum.PlayOff
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs]: Can we group all conditions under if(scheme==Playoff) ?

File: tests/VolleyManagement.UnitTests/Services/GameService/Fixture/GameServiceTestFixture.cs [hunk 2]
```diff
@@ -913,6 +913,54 @@ public GameServiceTestFixture TestMinimumOddTeamsPlayOffSchedule()
             return this;
         }
 
+        public GameServiceTestFixture TestTeamsEqualsPlayOffScheme()
```
Comment (sdiachen) [inline tests/VolleyManagement.UnitTests/Services/GameService/Fixture/GameServiceTestFixture.cs]: TeamsEqualsPlayoffScheme is hard to understand

Comment (MikolaB) [reply tests/VolleyManagement.UnitTests/Services/GameService/Fixture/GameServiceTestFixture.cs]: What you think about SameTeamsInOneGamePlayOffScheme

Comment (sdiachen) [reply tests/VolleyManagement.UnitTests/Services/GameService/Fixture/GameServiceTestFixture.cs]: NoTeamsGameInPlayoff

File: src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs [hunk 3]
```diff
@@ -469,8 +477,15 @@ private void ValidateGameInTournament(Game game, TournamentScheduleDto tournamen
                     TournamentId = tournamentSсheduleInfo.Id
                 });
 
-            var newGameDivisionId = teamsInTournament
-                .First(t => t.TeamId == newGame.AwayTeamId || t.TeamId == newGame.HomeTeamId).DivisionId;
+            var newGameDivisionId = (int?)0;
+
+            if (!(newGame.AwayTeamId == null && newGame.HomeTeamId == null && tournamentSсheduleInfo.Scheme == TournamentSchemeEnum.PlayOff))
```
Comment (sdiachen) [inline src/VolleyManagement.Backend/VolleyManagement.Services/GameService.cs]: this condition?

File: tests/VolleyManagement.UnitTests/Services/GameService/GameServiceTests.cs [hunk 4]
```diff
@@ -1646,43 +1646,85 @@ public void Edit_AddResultsToGameInPlayoff_NewGameIsScheduled()
...
-            List<Game> games = new GameTestFixture()
-                .TestMinimumEvenTeamsPlayOffSchedule()
+            var games = new GameTestFixture()
+                .SameTeamsInOneGamePlayOffScheme()
                 .Build();
 
-            List<GameResultDto> gameInfo = new GameServiceTestFixture()
-                .TestMinimumEvenEmptyGamesPlayoff()
+            var gameInfo = new GameServiceTestFixture()
+                .SameTeamsInOneGamePlayOffScheme()
```
Comment (YuriiChornyi) [inline tests/VolleyManagement.UnitTests/Services/GameService/GameServiceTests.cs]: Why var?

**Pull request merged**